qiskit/circuit/library/arithmetic/piecewise_chebyshev.py

```diff
@@ -0,0 +1,327 @@
+        Note that this may change the underlying quantum register, if the number of state qubits
+        changes.
+
+        Args:
+            polynomials: The new breakpoints for the piecewise approximation.
+        """
+        if self._polynomials is None or polynomials != self._polynomials:
+            self._invalidate()
+            self._polynomials = polynomials
+
+            self._reset_registers(self.num_state_qubits)
+
+    @property
+    def num_state_qubits(self) -> int:
+        r"""The number of state qubits representing the state :math:`|x\rangle`.
+
+        Returns:
+            The number of state qubits.
+        """
+        return self._num_state_qubits
+
+    @num_state_qubits.setter
+    def num_state_qubits(self, num_state_qubits: Optional[int]) -> None:
+        """Set the number of state qubits.
+
+        Note that this may change the underlying quantum register, if the number of state qubits
+        changes.
+
+        Args:
+            num_state_qubits: The new number of qubits.
+        """
+        if self._num_state_qubits is None or num_state_qubits != self._num_state_qubits:
+            self._invalidate()
+            self._num_state_qubits = num_state_qubits
+
+            # Set breakpoints if they haven't been set
+            if num_state_qubits is not None and self._breakpoints is None:
+                self.breakpoints = [0, 2 ** num_state_qubits]
+
+            self._reset_registers(num_state_qubits)
+
+    def _reset_registers(self, num_state_qubits: Optional[int]) -> None:
+        if num_state_qubits is not None:
+            qr_state = QuantumRegister(num_state_qubits)
+            qr_target = QuantumRegister(1)
+            self.qregs = [qr_state, qr_target]
+            self._ancillas = []
+            self._qubits = qr_state[:] + qr_target[:]
+
+            num_ancillas = num_state_qubits
+            if num_ancillas > 0:
+                qr_ancilla = AncillaRegister(num_ancillas)
+                self.add_register(qr_ancilla)
+
+        else:
+            self.qregs = []
+            self._qubits = []
+            self._ancillas = []
+
+    def _build(self):
+        """Build the circuit. The operation is considered successful when q_objective is :math:`|1>`
+        """
+        # do not build the circuit if _data is already populated
+        if self._data is not None:
+            return
+
+        self._data = []
+
+        # check whether the configuration is valid
+        self._check_configuration()
+
+        self._poly_r = PiecewisePolynomialPauliRotations(self.num_state_qubits,
+                                                         self.breakpoints, self.polynomials)
+
+        qr_state = self.qubits[:self.num_state_qubits]
+        qr_target = [self.qubits[self.num_state_qubits]]
+        qr_ancillas = self.qubits[self.num_state_qubits + 1:]
+
+        # Apply polynomial approximation
+        self.append(self._poly_r.to_instruction(), qr_state[:] + qr_target + qr_ancillas[:])
```

Comment:
Doesn't need to be a member anymore
```suggestion
        poly_r = PiecewisePolynomialPauliRotations(self.num_state_qubits,
                                                         self.breakpoints, self.polynomials)

        qr_state = self.qubits[:self.num_state_qubits]
        qr_target = [self.qubits[self.num_state_qubits]]
        qr_ancillas = self.qubits[self.num_state_qubits + 1:]

        # Apply polynomial approximation
        self.append(poly_r.to_instruction(), qr_state[:] + qr_target + qr_ancillas[:])
```